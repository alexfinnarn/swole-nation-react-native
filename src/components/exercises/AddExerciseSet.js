import React, {useState} from 'react';
import {Text, TextInput, View, Picker, Button} from "react-native";
import shortId from 'shortid';
import {styles} from "../Styles";

export default function AddExerciseSet({item = {key: "0", reps: 0, weight: 100}, exercise, toAdd, updater}) {
  const [reps, setReps] = useState(item.reps.toString());
  const [weight, setWeight] = useState(item.weight.toString());

  function updatePicker(value, type) {
    if (type === 'reps') {
      setReps(value);
      updater({reps: value, weight: weight, key: item.key}, exercise, shortId.generate(), 'UPDATE_SET');
    } else {
      setWeight(value);
      updater({reps: reps, weight: value, key: item.key}, exercise, shortId.generate(), 'UPDATE_SET');
    }
  }

  return (
    <View style={{flex: 1, flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', padding: 5, borderWidth: 1, borderColor: '#d6d7da', borderRadius: 4}}>
      <Text>Reps</Text>
      <Picker
        selectedValue={reps}
        style={{height: 60, width: 120}}
        onValueChange={(itemValue) => updatePicker(itemValue, 'reps')}>
        {[...Array(12)].map((e, index) => <Picker.Item key={index.toString()} label={index.toString()} value={index.toString()}/>)}
      </Picker>
      <Text>Weight</Text>
      <Picker
        selectedValue={weight}
        style={{height: 60, width: 120}}
        onValueChange={(itemValue) => updatePicker(itemValue, 'weight')}>
        {[...Array(75)].map((e, index) => <Picker.Item key={(index * 5).toString()} label={(index * 5).toString()} value={(index * 5).toString()}/>)}
      </Picker>
      {toAdd
        ? <Button onPress={() => {
          updater({reps: reps, weight: weight}, exercise, shortId.generate(), 'ADD_SET')
        }} title="+"/>
        : <Button onPress={() => {
          updater({key: item.key, reps: reps, weight: weight}, exercise, shortId.generate(), 'REMOVE_SET')
        }} title="X"/>
      }
    </View>

  );
}

