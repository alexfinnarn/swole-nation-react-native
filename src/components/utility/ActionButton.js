import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {home} from "../Styles";

export default function ActionButton({text, action, label, styles, disabled = false}) {
  const bgColor = disabled ? '#bababa' : '#21897E';

  return (
    <View style={[{flex: 1, flexDirection: 'column'}, styles]}>
      <TouchableOpacity
        disabled={disabled}
        accessible={true}
        accessibilityLabel={label}
        activeOpacity={0.6}
        style={[{backgroundColor: bgColor}, home.actionButton]}
        onPress={action}>
        <View style={{flex: 1, justifyContent: 'center'}}>
          <Text style={home.actionButtonText}>{text}</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}
