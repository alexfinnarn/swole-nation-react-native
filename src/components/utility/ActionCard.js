import React from 'react';
import {Text, View} from 'react-native';
import {home} from '../Styles';
import ActionButton from './ActionButton';

export default function ActionCard({children, actionComponent}) {
  return (
    <View style={home.sectionContainer}>
      <View style={home.sectionLeft}>{children}</View>
      {actionComponent}
    </View>
  );
}
