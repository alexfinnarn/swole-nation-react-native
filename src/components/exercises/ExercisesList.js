import React from 'react';
import {FlatList, Text, View} from "react-native";
import {home, styles} from '../Styles';
import ActionButton from "../utility/ActionButton";
import ActionCard from "../utility/ActionCard";

export default function ExercisesList({ exercises, navigation, handle }) {

  return (
    <View style={styles.container} testID="exercises-list-root">
      <View style={{flex:9}}>
        <FlatList
          data={exercises}
          renderItem={(item) => <ExercisesListItem exercise={item}/>}
        />
      </View>
      <ActionButton text="Add Exercise" action={() => {
        handle.createExercise();
        navigation.navigate('AddExercise');
      }} />
    </View>
  );

  function ExercisesListItem({exercise}) {
    return (
      <ActionCard actionComponent={
        <ActionButton text="Edit" action={() => {
          handle.setActiveExerciseKey(exercise.item.key);
          navigation.navigate('AddExercise', {exerciseKey: exercise.item.key})
        }}/>
      }>
        <Text style={home.sectionHeaderText}>{exercise.item.name}</Text>
        <Text style={{flex: 4, marginTop: 15}}>Other content</Text>
      </ActionCard>
    );
  }
}



