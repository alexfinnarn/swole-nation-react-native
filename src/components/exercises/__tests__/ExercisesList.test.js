import React from 'react';
import ExercisesList from '../ExercisesList';
import { render, fireEvent } from 'react-native-testing-library';
import data from '../../../store/data';
import ActionCard from "../../utility/ActionCard";
import ActionButton from "../../utility/ActionButton";
import {TouchableOpacity} from "react-native";
import AddExerciseSet from "../AddExerciseSet";

let renderer, instance ={};
let exercises = Object.keys(data.exercises).map(key => data.exercises[key]);
const handle = {
  createExercise: jest.fn(),
  setActiveExerciseKey: jest.fn(),
};
const navigation = {
  navigate: jest.fn()
};

beforeEach(() => {
  renderer = render(<ExercisesList exercises={exercises} navigation={navigation} handle={handle} />);
  instance = renderer.getByTestId('exercises-list-root');
}, 0);

describe('<ExercisesList />', () => {
  it('Renders correctly', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('Renders all 10 exercise action cards on load', () => {
    const actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(10);
    expect(renderer.getAllByText('Edit').length).toBe(10);

    // First card should have an action button.
    expect(actionCards[0].props.actionComponent.type).toEqual(ActionButton);
  });

  it('Navigates to exercise scree with correct exercise ID', () => {
    const navigationButtons = instance.findAllByType(TouchableOpacity);

    // Navigate to Squats Warmup.
    fireEvent(navigationButtons[0], 'press');
    expect(navigation.navigate).toHaveBeenCalledWith("AddExercise", {
      exerciseKey: "BwEBDwoFBgQ",
      title: "Squats Warmup"
    });

    // Navigate to Deadlifts.
    fireEvent(navigationButtons[9], 'press');
    expect(navigation.navigate).toHaveBeenLastCalledWith("AddExercise", {
      exerciseKey: "DAYCDAQKAw4",
      title: "Deadlifts"
    });

    // Navigate to New Exercise.
    // fireEvent(renderer.getByText('Add Exercise'), 'press');
    // expect(navigation.navigate).toHaveBeenCalledWith("AddExercise");
    // expect(handle.createExercise).toHaveBeenCalled();

  });

  it('Updates FlatList when re-rendered', () => {
    let navigationButtons = instance.findAllByType(TouchableOpacity);

    // Navigate to Squats.
    fireEvent(navigationButtons[0], 'press');
    expect(navigation.navigate).toHaveBeenCalledWith("AddExercise", {
      exerciseKey: "BwEBDwoFBgQ",
      title: "Squats Warmup"
    });

    // Rearrange Deadlifts to take Squats place in array.
    const squats = exercises.shift();
    const deadlifts = exercises.pop();
    exercises.unshift(deadlifts);

    renderer = renderer.update(<ExercisesList exercises={exercises} navigation={navigation} handle={handle} />);
    navigationButtons = instance.findAllByType(TouchableOpacity);

    expect(instance.findAllByType(ActionCard).length).toBe(9);

    // Navigate to Deadlifts.
    fireEvent(navigationButtons[0], 'press');
    expect(navigation.navigate).toHaveBeenLastCalledWith("AddExercise", {
      exerciseKey: "DAYCDAQKAw4",
      title: "Deadlifts"
    });
  });
});
