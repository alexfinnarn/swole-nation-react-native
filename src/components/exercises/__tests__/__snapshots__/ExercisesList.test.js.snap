// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExercisesList /> Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "flexDirection": "column",
      "paddingTop": 5,
    }
  }
  testID="exercises-list-root"
>
  <View
    style={
      Object {
        "flex": 9,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "image": Object {
              "testUri": "../../../assets/squats.gif",
            },
            "instructions": "Do with a barbell",
            "key": "BwEBDwoFBgQ",
            "name": "Squats Warmup",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 45,
              },
              Object {
                "reps": 5,
                "weight": 45,
              },
              Object {
                "reps": 5,
                "weight": 95,
              },
              Object {
                "reps": 5,
                "weight": 135,
              },
              Object {
                "reps": 3,
                "weight": 185,
              },
              Object {
                "reps": 2,
                "weight": 205,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/squats.gif",
            },
            "instructions": "Do with a barbell",
            "key": "Bw0ECwAKBws",
            "name": "Squats",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 230,
              },
              Object {
                "reps": 5,
                "weight": 230,
              },
              Object {
                "reps": 5,
                "weight": 230,
              },
              Object {
                "reps": 5,
                "weight": 230,
              },
              Object {
                "reps": 5,
                "weight": 230,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/bench-press.gif",
            },
            "instructions": "Do with a barbell",
            "key": "BwwBDAIJDgw",
            "name": "Bench Press Warmup",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 45,
              },
              Object {
                "reps": 5,
                "weight": 45,
              },
              Object {
                "reps": 5,
                "weight": 95,
              },
              Object {
                "reps": 3,
                "weight": 135,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/bench-press.gif",
            },
            "instructions": "Do with a barbell",
            "key": "DgcBCA8BCQo",
            "name": "Bench Press",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 165,
              },
              Object {
                "reps": 5,
                "weight": 165,
              },
              Object {
                "reps": 5,
                "weight": 165,
              },
              Object {
                "reps": 5,
                "weight": 165,
              },
              Object {
                "reps": 5,
                "weight": 165,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/shoulder-press.gif",
            },
            "instructions": "Do with a barbell",
            "key": "DwcACw4PCgI",
            "name": "Overhead Press Warmup",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 45,
              },
              Object {
                "reps": 5,
                "weight": 45,
              },
              Object {
                "reps": 3,
                "weight": 75,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/shoulder-press.gif",
            },
            "instructions": "Do with a barbell",
            "key": "CAMBAwsFBwQ",
            "name": "Overhead Press",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 110,
              },
              Object {
                "reps": 5,
                "weight": 110,
              },
              Object {
                "reps": 5,
                "weight": 110,
              },
              Object {
                "reps": 5,
                "weight": 110,
              },
              Object {
                "reps": 5,
                "weight": 110,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/barbell-row.gif",
            },
            "instructions": "Do with a barbell",
            "key": "BwYGAw0NDwo",
            "name": "Barbell Row Warmup",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 95,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/barbell-row.gif",
            },
            "instructions": "Do with a barbell",
            "key": "BwQKAwYKAgA",
            "name": "Barbell Row",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 115,
              },
              Object {
                "reps": 5,
                "weight": 115,
              },
              Object {
                "reps": 5,
                "weight": 115,
              },
              Object {
                "reps": 5,
                "weight": 115,
              },
              Object {
                "reps": 5,
                "weight": 115,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/deadlift.gif",
            },
            "instructions": "Do with a barbell",
            "key": "AQcNAQkBCwo",
            "name": "Deadlifts Warmup",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 135,
              },
              Object {
                "reps": 5,
                "weight": 185,
              },
              Object {
                "reps": 5,
                "weight": 205,
              },
            ],
          },
          Object {
            "image": Object {
              "testUri": "../../../assets/deadlift.gif",
            },
            "instructions": "Do with a barbell",
            "key": "DAYCDAQKAw4",
            "name": "Deadlifts",
            "sets": Array [
              Object {
                "reps": 5,
                "weight": 225,
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Squats Warmup
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Squats
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Bench Press Warmup
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Bench Press
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Overhead Press Warmup
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Overhead Press
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Barbell Row Warmup
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Barbell Row
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Deadlifts Warmup
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Deadlifts
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Do with a barbell
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
